cmd/karavictl/cmd/storage_create.go

```diff
@@ -191,41 +209,115 @@ var storageCreateCmd = &cobra.Command{
 		// TODO(ian): This logic should ideally be performed remotely, not
 		// in the client.
 
-		sioClient, err := goscaleio.NewClientWithArgs(epURL.String(), "", true, false)
-		if err != nil {
-			errAndExit(err)
-		}
+		var tempStorage SystemType
 
-		_, err = sioClient.Authenticate(&goscaleio.ConfigConnect{
-			Username: input.User,
-			Password: input.Password,
-		})
-		if err != nil {
-			errAndExit(err)
-		}
+		switch input.Type {
+		case powerflex:
+			tempStorage = storage[powerflex]
+			if tempStorage == nil {
+				tempStorage = make(map[string]System)
+			}
 
-		resp, err := sioClient.FindSystem(input.SystemID, "", "")
-		if err != nil {
-			errAndExit(err)
-		}
-		if resp.System.ID != input.SystemID {
-			fmt.Fprintf(cmd.ErrOrStderr(), "system id %q not found", input.SystemID)
-			osExit(1)
+			fmt.Printf("Connecting to PowerFlex system...\n")
+			sioClient, err := goscaleio.NewClientWithArgs(epURL.String(), "", true, false)
+			if err != nil {
+				errAndExit(err)
+			}
+
+			_, err = sioClient.Authenticate(&goscaleio.ConfigConnect{
+				Username: input.User,
+				Password: input.Password,
+			})
+			if err != nil {
+				errAndExit(err)
+			}
+
+			if len(input.SystemID) == 0 {
+				errAndExit(errSystemIDNotSpecified)
+			}
+
+			resp, err := sioClient.FindSystem(input.SystemID[0], "", "")
+			if err != nil {
+				errAndExit(err)
+			}
+			if resp.System.ID != input.SystemID[0] {
+				fmt.Fprintf(cmd.ErrOrStderr(), "system id %q not found", input.SystemID)
+				osExit(1)
+			}
+
+			storageID := strings.Trim(SystemID{Value: input.SystemID[0]}.String(), "\"")
+			tempStorage[storageID] = System{
+				User:     input.User,
+				Password: input.Password,
+				Endpoint: input.Endpoint,
+				Insecure: input.Insecure,
+			}
+
+		case powermax:
+			tempStorage = storage[powermax]
+			if tempStorage == nil {
```

Comment:
Duplicates the same nil check in the powerflex case - perhaps just initialize this before the `switch`?

Reply:
Resolved